mmf/models/vilbert.py

```diff
@@ -1186,9 +1192,13 @@ def forward(
         if self.training_head_type == "nlvr2":
             pooled_output = pooled_output.view(-1, pooled_output.size(1) * 2)
 
-        logits = self.classifier(pooled_output)
-        reshaped_logits = logits.contiguous().view(-1, self.num_labels)
-        output["scores"] = reshaped_logits
+        if self.training_head_type == 'visual7w':
+            logits = self.classifier(self.dropout(sequence_output_v))
+            output["scores"] = logits[:, -4:].squeeze()
```

Comment:
Let's move this to config and use something like `num_multiple_choices` etc instead of hardcoding `4`.

Reply:
Done